testsuite/standalone/src/test/java/org/wildfly/extras/creaper/core/online/SslOnlineManagementClientTest.java

```diff
@@ -0,0 +1,157 @@
+    }
+
+    @Test
+    @InSequence(Integer.MAX_VALUE)
+    public void stopServer() throws IOException, CommandFailedException {
+        this.controller.stop(ManualTests.ARQUILLIAN_CONTAINER_MGMT_PROTOCOL_REMOTE);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    @Override
+    public void clientConfiguredForDomain() throws IOException {
+        ManagementClient.online(OnlineOptions.domain().build().localDefault().ssl(sslOptions).build());
+    }
+
+
+    private static void prepareKeyAndTrustStoreFiles() throws GeneralSecurityException, IOException {
+        final KeyPairAndCertificate serverKeyPair = KeyPairAndCertificate.generateSelfSigned("test-creaper-server");
+        serverKeyStoreFile = serverKeyPair.toTmpKeyStoreFile(SERVER_KEY_ALIAS, SERVER_KEY_PASSWORD);
+        clientTrustStoreFile = serverKeyPair.toTmpTrustStoreFile(SERVER_CERTIFICATE_ALIAS, SERVER_CERTIFICATE_PASSWORD);
+
+        final KeyPairAndCertificate clientKeyPair = KeyPairAndCertificate.generateSelfSigned("test-creaper-client");
+        clientKeyStoreFile = clientKeyPair.toTmpKeyStoreFile(CLIENT_KEY_ALIAS, CLIENT_KEY_PASSWORD);
+        serverTrustStoreFile = clientKeyPair.toTmpTrustStoreFile(CLIENT_CERTIFICATE_ALIAS, CLIENT_CERTIFICATE_PASSWORD);
+    }
+
+    private static void prepareSslOptions() {
+        sslOptions = new SslOptions.Builder()
+                .keyStore(clientKeyStoreFile)
+                .keyStorePassword(CLIENT_KEY_PASSWORD)
+                .key(CLIENT_KEY_ALIAS, CLIENT_KEY_PASSWORD)
+                .trustStore(clientTrustStoreFile)
+                .trustStorePassword(SERVER_CERTIFICATE_PASSWORD)
+                .build();
+    }
+
+    private static void configureServerManagementInterfaces() throws CommandFailedException {
+        AddHttpsManagementSecurityRealm addHttpsSecurityRealm = new AddHttpsManagementSecurityRealm
+                .Builder(SECURITY_REALM)
+                .keystorePath(serverKeyStoreFile.getAbsolutePath())
+                .keystorePassword(SERVER_KEY_PASSWORD)
+                .keystoreProvider(KeyStoreType.DEFAULT_TYPE.typeName().toUpperCase())
+                .keyAlias(SERVER_KEY_ALIAS)
+                .truststorePath(serverTrustStoreFile.getAbsolutePath())
+                .truststorePassword(CLIENT_CERTIFICATE_PASSWORD)
+                .truststoreProvider(KeyStoreType.DEFAULT_TYPE.typeName().toUpperCase())
+                .build();
+
+        AddNativeManagementInterface addNativeInterface = new AddNativeManagementInterface
+                .Builder("ManagementRealm", "management-http").replaceExisting().build();
+
+        AddHttpManagementInterface addHttpsInterface = new AddHttpManagementInterface.Builder(SECURITY_REALM)
+                .secureSocketBinding("management-https").httpUpgradeEnabled(true).replaceExisting().build();
+
+        offlineClient.apply(addHttpsSecurityRealm, addNativeInterface, addHttpsInterface);
```

Comment:
I have never seen anyone using this :-)

(Nothing wrong with it. No need to change. Just an observation :-) )